src/emailservice/email_server.py

```diff
@@ -143,7 +144,24 @@ def start(dummy_mode):
   except KeyboardInterrupt:
     server.stop(0)
 
+def InitProfiler():
+  for retry in range(3):
+    try:
+      googlecloudprofiler.start(service='email_server', service_version='1.0.0', verbose=0)
+      logger.info("Successfully started Stackdriver Profiler.")
+      return
+    except (ValueError, NotImplementedError) as exc:
+      logger.info("Unable to start Stackdriver Profiler Python agent in email_server.py.\n" + str(exc))
+      if (retry < 3):
+        logger.info("Sleeping %d to retry initializing Stackdriver Profiler"%(retry*10))
+        time.sleep (retry *10)
```

Comment:
10 is too much. 1-1.5s should be fine here.

Reply:
I agree. However, I tried to follow the example from the shippingservice code snippet you linked above which falls back 10, 20 & 30 seconds. Happy to change it to 1 second.

Reply:
No you're right. Let's keep 10s. Funny enough we're even seeing it failing consistently on  Istio 1.0.3 (#183) but then eventually succeed. Also I realized since that RPCs are still handled during this period (this registration doesn't block).